lib/sonar/cli/cli.rb

```diff
@@ -27,7 +29,12 @@ def usage
 
     desc 'search_all [QUERY TERM]', 'Search all Sonar record types'
     def search_all(term)
-      Sonar::Search::QUERY_TYPES_MAP.keys.each do |type|
+      if term =~ IS_IP
+        types = ip_search_type_names
+      else
+        types = domain_search_type_names
+      end
+      types.each do |type|
         search(type, term) rescue 'Search failed'
```

Comment:
IMO, unless the output somehow indicates what results are for which "sub" search, searches with empty results shouldn't print anything out.  I'm not sure what it would take to fix this:

```
$  ./bin/sonar search_all 4.2.2.1
{"collection":[{"address":"4.2.2.1","name":"a.resolvers.level3.net","sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]},{"address":"4.2.2.100","name":"tac-pri.sys.gtei.net","sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]},{"address":"4.2.2.101","name":"tac-bak.sys.gtei.net","sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]},{"address":"4.2.2.12","name":"test-record.level3.net","sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]}],"more":false,"warning":"","iterator_id":"479078fb97b9e71d"}
{"collection":[],"more":false,"warning":"","iterator_id":"a17def43c9e351f1"}
{"collection":[],"more":false,"warning":"","iterator_id":"041d5e0d6b1bdbac"}
{"collection":[{"key":"4.2.2.1","value":{"ip":"4.2.2.1","data.size":75,"data.dns_version":"Version: recursive-main/18640994","ip.country_code":"US","ip.country_code3":"USA","ip.country_name":"United States","ip.latitude":"37.750999450683594","ip.longitude":"-97.8219985961914","ip.org":"Level 3 Communications"},"scan_type":["udp-dns"],"sources":["20170605","20170703"]}],"more":false,"warning":"","iterator_id":"84a9f96577b68f77"}
{"collection":[{"key":"4.2.2.1","value":"a.resolvers.level3.net","scan_type":["rdns"],"sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]},{"key":"4.2.2.100","value":"tac-pri.sys.gtei.net","scan_type":["rdns"],"sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]},{"key":"4.2.2.101","value":"tac-bak.sys.gtei.net","scan_type":["rdns"],"sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]},{"key":"4.2.2.12","value":"test-record.level3.net","scan_type":["rdns"],"sources":["20170517","20170524","20170531","20170607","20170614","20170621","20170630","20170705","20170712","20170719","20170726","20170802"]}],"more":false,"warning":"","iterator_id":"82f3a67363c3504c"}
{"error":"Data not found","errors":["whois data is no longer collected"]}
```